### PR TITLE
New Resource: `azurerm_arc_private_link_scope`

### PR DESCRIPTION
**Pull request by liuwuliuyun**
New Resource: `azurerm_arc_private_link_scope`

Swagger Link:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/stable/2022-11-10/privateLinkScopes.json

Tesing Evidence:

```
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.1\tmp\GoLand\___TestAccArcPrivateLinkScope_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_hybridcompute.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/hybridcompute #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.1\tmp\GoLand\___TestAccArcPrivateLinkScope_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_hybridcompute.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccArcPrivateLinkScope_basic\E$ #gosetup
=== RUN   TestAccArcPrivateLinkScope_basic
=== PAUSE TestAccArcPrivateLinkScope_basic
=== CONT  TestAccArcPrivateLinkScope_basic
--- PASS: TestAccArcPrivateLinkScope_basic (170.70s)
PASS

Process finished with the exit code 0
```

```
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.1\tmp\GoLand\___TestAccArcPrivateLinkScope_update_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_hybridcompute.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/hybridcompute #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.1\tmp\GoLand\___TestAccArcPrivateLinkScope_update_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_hybridcompute.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccArcPrivateLinkScope_update\E$ #gosetup
=== RUN   TestAccArcPrivateLinkScope_update
=== PAUSE TestAccArcPrivateLinkScope_update
=== CONT  TestAccArcPrivateLinkScope_update
--- PASS: TestAccArcPrivateLinkScope_update (245.81s)
PASS


Process finished with the exit code 0
```

### CODE REVIEW

File: internal/services/hybridcompute/arc_private_link_scope_resource.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+func (a ArcPrivateLinkScopeResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.HybridCompute.PrivateLinkScopesClient
+
+			id, err := privatelinkscopes.ParseProviderPrivateLinkScopeID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing id: %+v", err)
```
Comment (stephybun) [inline internal/services/hybridcompute/arc_private_link_scope_resource.go:170]:
```suggestion
				return err
```

File: internal/services/hybridcompute/arc_private_link_scope_resource.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+			}
+
+			if model := existing.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				if model.Tags != nil {
+					state.Tags = *model.Tags
+				}
+				if model.Properties != nil && model.Properties.PublicNetworkAccess != nil {
+					state.PublicNetworkAccessEnabled = *model.Properties.PublicNetworkAccess == privatelinkscopes.PublicNetworkAccessTypeEnabled
+				}
```
Comment (stephybun) [inline internal/services/hybridcompute/arc_private_link_scope_resource.go:193]:
```suggestion
				state.Tags = pointer.From(model.Tags)

				publicNetworkAccess := false
				if props := model.Properties; props != nil && props.PublicNetworkAccess != nil {
					publicNetworkAccess = *model.Properties.PublicNetworkAccess == privatelinkscopes.PublicNetworkAccessTypeEnabled
				}
				state.PublicNetworkAccessEnabled = publicNetworkAccess
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @liuwuliuyun, a couple of comments that need to be fixed up but otherwise this looks fine.

File: internal/services/hybridcompute/arc_private_link_scope_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+resource "azurerm_arc_private_link_scope" "test" {
+  name                = "acctestPLS-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ArcPrivateLinkScopeResource) complete(data acceptance.TestData) string {
```
[the remark this answers is not in the document]
Comment (liuwuliuyun) [reply internal/services/hybridcompute/arc_private_link_scope_resource_test.go:99]: Sure, I will add as suggested.

**Comment by liuwuliuyun** [comment]
```
Generating script.
Script contents:
TF_ACC=1 go test -parallel 10 -v ./internal/services/hybridcompute -run=TestAccArcPrivateLinkScope_ -timeout '720m' -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
========================== Starting Command Output ===========================
/usr/bin/bash /mnt/vss/_work/_temp/e5bb1e15-e9b4-45f8-bde0-55f9fe7d9de6.sh
=== RUN   TestAccArcPrivateLinkScope_basic
=== PAUSE TestAccArcPrivateLinkScope_basic
=== RUN   TestAccArcPrivateLinkScope_complete
=== PAUSE TestAccArcPrivateLinkScope_complete
=== RUN   TestAccArcPrivateLinkScope_update
=== PAUSE TestAccArcPrivateLinkScope_update
=== RUN   TestAccArcPrivateLinkScope_requireImport
=== PAUSE TestAccArcPrivateLinkScope_requireImport
=== CONT  TestAccArcPrivateLinkScope_basic
=== CONT  TestAccArcPrivateLinkScope_update
=== CONT  TestAccArcPrivateLinkScope_complete
=== CONT  TestAccArcPrivateLinkScope_requireImport
--- PASS: TestAccArcPrivateLinkScope_complete (85.63s)
--- PASS: TestAccArcPrivateLinkScope_requireImport (118.00s)
--- PASS: TestAccArcPrivateLinkScope_basic (127.20s)
--- PASS: TestAccArcPrivateLinkScope_update (152.93s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/hybridcompute	152.954s
Finishing: Run test in parallel

```

File: website/docs/r/arc_private_link_scope.html.markdown [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## Timeouts
+
+The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/language/resources/syntax#operation-timeouts) for certain actions:
+
+* `create` - (Defaults to 30 minutes) Used when creating the Hybrid Compute.
+* `read` - (Defaults to 5 minutes) Used when retrieving the Hybrid Compute.
+* `update` - (Defaults to 30 minutes) Used when updating the Hybrid Compute.
+* `delete` - (Defaults to 30 minutes) Used when deleting the Hybrid Compute.
+
```
Comment (stephybun) [inline website/docs/r/arc_private_link_scope.html.markdown:58]: Hybrid Compute is still referenced in all of the arguments and the timeouts - please fix these

```suggestion
* `name` - (Required) The name which should be used for the Azure Arc Private Link Scope. Changing this forces a new Hybrid Compute to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the Azure Arc Private Link Scope should exist. Changing this forces a new Hybrid Compute to be created.

---

* `public_network_access_enabled` - (Optional) Indicates whether machines associated with the private link scope can also use public Azure Arc service endpoints. Defaults to `false`. Possible values are `true` and `false`.

* `tags` - (Optional) A mapping of tags which should be assigned to the Azure Arc Private Link Scope.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported: 

* `id` - The ID of the Azure Arc Private Link Scope.

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/language/resources/syntax#operation-timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Hybrid Compute.
* `read` - (Defaults to 5 minutes) Used when retrieving the Hybrid Compute.
* `update` - (Defaults to 30 minutes) Used when updating the Hybrid Compute.
* `delete` - (Defaults to 30 minutes) Used when deleting the Hybrid Compute.

```

**Review by stephybun — CHANGES_REQUESTED** [review]

File: website/docs/r/arc_private_link_scope.html.markdown (hunk 4 above)
Comment (liuwuliuyun) [reply website/docs/r/arc_private_link_scope.html.markdown]: Fixed as suggested.

**Review by stephybun — APPROVED** [review]
Thanks @liuwuliuyun LGTM 👍

**Pull request merged**